### PR TITLE
Issue #14937: Migrated section 5.2.2 Class Names to Chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

File: config/checkstyle-resources-suppressions.xml [hunk 1]
```diff
@@ -81,7 +81,7 @@
   <suppress checks="OuterTypeFilename"
     files="[\\/]InputOuterType.*\.java"/>
   <suppress checks="OuterTypeFilename"
-    files="rule522typenames[\\/]InputTypeName\.java"/>
+    files="rule522typenames[\\/]InputClassNames\.java"/>
```
Comment (Zopsss) [inline config/checkstyle-resources-suppressions.xml:84]: Renamed the input file names so had to make changes here

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package com.google.checkstyle.test.chapter5naming.rule522typenames;
+
+class inputHeaderClass {
+    // violation above 'Type name 'inputHeaderClass' must match pattern '.*[A-Z][a-zA-Z0-9].*'.'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java:4]: Original pattern to match was `^[A-Z][a-zA-Z0-9]*$`

But it was causing error while verifying the message so I had to modify a little bit.

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java:4]: please remvove '.*[A-Z][a-zA-Z0-9].*' at all no need to keep it, 
just `// violation above 'Type name 'inputHeaderClass' must match pattern`

**Review by romani — CHANGES_REQUESTED** [review]
see above

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    // violation above 'Type name 'inputHeaderClass' must match pattern .*'
+
+    public interface InputHeader___Interface {};
+    // violation above 'Type name 'InputHeader___Interface' .* .*'
+
+    public enum inputHeaderEnum { one, two };
+    // violation above 'Type name 'inputHeaderEnum' must match pattern .*'
+
+    class NoValid$Name {}
+    // violation above 'Type name '.*' must match pattern .*'
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java:13]: Remove trailing .* It should not be required

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java (hunk 3 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java]: done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**